### PR TITLE
Deprecate `vocab_size` in other two VLMs

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Continuing on #31655, let's deprecate the `model.config.vocab_size` for all VLMs for consistency across repo and use `model.config.text_config.vocab_size` in future models.

Currently only Fuyu and Paligemma have `vocab_size` in config file, and I depreacted it to v4.43. Or should it be v4.44?

### CODE REVIEW

File: src/transformers/models/fuyu/configuration_fuyu.py [hunk 1]
```diff
@@ -206,3 +208,20 @@ def _rope_scaling_validation(self):
             )
         if rope_scaling_factor is None or not isinstance(rope_scaling_factor, float) or rope_scaling_factor <= 1.0:
             raise ValueError(f"`rope_scaling`'s factor field must be a float > 1, got {rope_scaling_factor}")
+
+    @property
+    def vocab_size(self):
+        warnings.warn(
+            "The `vocab_size` attribute is deprecated and will be removed in v4.43, Please use `text_config.vocab_size` instead.",
```
Comment (amyeroberts) [inline src/transformers/models/fuyu/configuration_fuyu.py:215]: tiny nit - we should do this over two releases 

```suggestion
            "The `vocab_size` attribute is deprecated and will be removed in v4.44, Please use `text_config.vocab_size` instead.",
```

**Review by amyeroberts — APPROVED** [review]
Very nice - thanks for handling!

File: src/transformers/models/fuyu/configuration_fuyu.py (hunk 1 above)
Comment (zucchini-nlp) [reply src/transformers/models/fuyu/configuration_fuyu.py:215]: oke, will change for 4.44 in paligemma also

**Comment by prakash-shubham-2003** [comment]
@zucchini-nlp @amyeroberts 
After this update, I am facing an error while making an instance of LlavaProcessor. This is the code I wrote:
'processor = AutoProcessor.from_pretrained("llava-hf/llava-1.5-7b-hf")'

It gives the following error:
'TypeError: LlavaProcessor.__init__() got an unexpected keyword argument 'chat_template''

It was working fine before the update. I have the following versions of transformers and bitsandbytes installed:
'!pip install -q -U transformers==4.37.2'
'!pip install -q bitsandbytes==0.41.3 accelerate==0.25.0'

Is there any issue with the current update or is there any other solution for this problem?

**Comment by zucchini-nlp** [comment]
Answered the question in another thread. Please open an issue if you still encounter the error after updating transformers :)

**Pull request merged**